### PR TITLE
script to manually create a phylo run workflow

### PR DESCRIPTION
**Pull request by ttung**
### Description
In the future, this will be created by a web endpoint, but to build out the rest of the backend, I need a script that does this for now.

Depends on #236

#### Issue
[ch126782](https://app.clubhouse.io/genepi/story/126782/workflow-to-pull-the-data-from-aspen-api-endpoint-and-launch-nextstrain-build)

### Test plan
created a workflow, manual inspection seems to suggest that it was reasonable.

### CODE REVIEW

File: .hgtags [hunk 1]
```diff
@@ -0,0 +1,6 @@
+c0cda8b0ffb4a4c71feabfdbf01c4aeb2af27157 t1
+0f598f1ff40c91da04278989f1e4c68d00bc87e0 t1
```
Comment (shanaxel42) [inline .hgtags]: rebase badness?

File: src/py/aspen/cli/db.py [hunk 2]
```diff
@@ -108,3 +128,115 @@ def import_covidhub_project(
...
+    with session_scope(engine) as session:
+        group = session.query(Group).filter(Group.name == group_name).one()
+
+        all_samples: Iterable[Sample] = (
+            session.query(Sample)
+            .filter(
+                or_(
+                    Sample.public_identifier.in_(samples),
+                    and_(
+                        expression.true()
```
Comment (shanaxel42) [inline src/py/aspen/cli/db.py:247]: What's getting checked here?

File: src/py/aspen/cli/db.py [hunk 3]
```diff
@@ -108,3 +128,115 @@ def import_covidhub_project(
...
+            aligned_gisaid_dump = (
+                session.query(AlignedGisaidDump)
+                .filter(AlignedGisaidDump.entity_id == aligned_gisaid_dump_id)
+                .one()
+            )
+        else:
+            aligned_gisaid_dump = (
+                session.query(AlignedGisaidDump)
+                .join(AlignedGisaidDump.producing_workflow)
+                .order_by(Workflow.end_datetime.desc())
```
Comment (shanaxel42) [inline src/py/aspen/cli/db.py:296]: oh chill, so it just get's the most recent one?

**Review by shanaxel42 — APPROVED** [review]
Exciting! Just a couple questions

File: src/py/aspen/cli/db.py [hunk 4]
```diff
@@ -168,3 +188,115 @@ def import_covidhub_project(
...
+            start_datetime=datetime.datetime.now(),
+            workflow_status=WorkflowStatusType.STARTED,
+            software_versions={},
+            group=group,
+            # TODO: need build template / build args.
+        )
+        workflow.inputs = list(pathogen_genomes)
+        workflow.inputs.append(aligned_gisaid_dump)
+        workflow.template_file_path = (
+            "workflows/setup_nextstrain_run/builds_template.yaml"
```
Comment (jgadling) [inline src/py/aspen/cli/db.py]: Where is this file?

File: src/py/aspen/cli/db.py (hunk 2 above)
Comment (ttung) [reply src/py/aspen/cli/db.py:247]: A sample is included if one of the following is true:

1. The sample is explicitly requested using `--sample`
2. `--all-group-samples` is specified AND the sample belongs to the group.

File: src/py/aspen/cli/db.py (hunk 3 above)
Comment (ttung) [reply src/py/aspen/cli/db.py:296]: yup.

File: src/py/aspen/cli/db.py (hunk 4 above)
Comment (ttung) [reply src/py/aspen/cli/db.py]: The workflow uses this file + the build args to generate the builds file.  It should be present in the target git refspec.  I have been iterating on the workflow in [this branch](https://github.com/chanzuckerberg/aspen/tree/ttung/setup-nextstrain-run).

File: src/py/aspen/cli/db.py [hunk 5]
```diff
@@ -168,3 +187,150 @@ def import_covidhub_project(
...
+        containerOverrides={
+            "command": [
+                git_refspec,
+                "src/py/workflows/nextstrain_run/setup.sh",
+                str(workflow_id),
+            ],
+            "vcpus": 32,
+            "memory": 370000,
+        }
+    )
```
Comment (ttung) [inline src/py/aspen/cli/db.py]: I added code to launch the batch job here.

File: src/py/aspen/cli/db.py (hunk 4 above)
Comment (ttung) [reply src/py/aspen/cli/db.py]: [here](https://github.com/chanzuckerberg/aspen/blob/ttung/setup-nextstrain-run/src/py/workflows/nextstrain_run/builds_template.yaml) in this case.

**Pull request merged**